beacon-chain/sync/subscriber.go

```diff
@@ -175,6 +176,7 @@ func (r *Service) subscribe(topic string, validator pubsub.Validator, handle sub
 // appropriate counter if the particular message fails to validate.
 func wrapAndReportValidation(topic string, v pubsub.Validator) (string, pubsub.Validator) {
 	return topic, func(ctx context.Context, pid peer.ID, msg *pubsub.Message) bool {
+		defer messagehandler.HandlePanic(ctx, msg)()
```

Comment:
There doesn't seem to be a need for a function to be returned. 

How about 
```go
defer messagehanlder.HandlePanic(ctx, msg)
```
?